src/civetweb.c

```diff
@@ -8076,14 +8076,19 @@ read_websocket(struct mg_connection *conn,
 	mg_set_thread_name("worker");
 }
 
-int
-mg_websocket_write(struct mg_connection *conn,
-                   int opcode,
-                   const char *data,
-                   size_t dataLen)
+static int
+mg_websocket_write_exec(struct mg_connection *conn,
+                        int opcode,
+                        const char *data,
+                        size_t dataLen,
+                        char mask)
 {
-	unsigned char header[10];
+	unsigned char header[14];
 	size_t headerLen = 1;
+	uint8_t masking_key[4];
+	unsigned char masked_data[dataLen];
```

Comment:
This does not work for Windows.
I get the following errors from the AppVeyor continuous integration (CI) environment:

https://ci.appveyor.com/project/civetweb/civetweb/build/232/job/aia6afxrslxtqpf3

error C2057: expected constant expression
error C2466: cannot allocate an array of constant size 0
error C2133: 'masked_data' : unknown size

This is probably some non-standard compiler extension, and it's very dangerous anyway: if masked_data is allocated on the stack (alloca) and dataLen is two billion the stack will overflow causing a crash of the whole program.